v0/destinations/freshsales/utils.js

```diff
@@ -0,0 +1,131 @@
+  let userSalesAccountResponse = await httpPOST(
+    endPoint,
+    userPayload,
+    requestOptions
+  );
+  userSalesAccountResponse = processAxiosResponse(userSalesAccountResponse);
+  if (userSalesAccountResponse.status !== 200) {
+    const errMessage =
+      userSalesAccountResponse.response.errors &&
+      userSalesAccountResponse.response.errors.message
+        ? userSalesAccountResponse.response.errors.message
+        : "";
+    const errorStatus =
+      userSalesAccountResponse.response.errors &&
+      userSalesAccountResponse.response.errors.code
+        ? userSalesAccountResponse.response.errors.code
+        : "500";
+    throw new CustomError(
+      `failed to fetch user accountDetails ${errMessage}`,
+      errorStatus
+    );
+  }
+  return userSalesAccountResponse;
+};
+
+/*
+ * This functions is used for getting Account details.
+ * If account is not exists then it will create it otherwise it will update it.
+ * @param {*} payloadBody
+ * @param {*} Config
+ * @returns
+ * ref: https://developers.freshworks.com/crm/api/#upsert_an_account
+ */
+const getAccountDetails = async (payloadBody, Config) => {
+  const requestOptions = {
+    headers: {
+      Authorization: `Token token=${Config.apiKey}`,
+      "Content-Type": "application/json"
+    }
+  };
+  const endPoint = `https://${Config.domain}${CONFIG_CATEGORIES.GROUP.endpoint}`;
+  let accountResponse = await httpPOST(endPoint, payloadBody, requestOptions);
+  accountResponse = processAxiosResponse(accountResponse);
+  if (accountResponse.status !== 200) {
+    const errMessage =
+      accountResponse.response.errors && accountResponse.response.errors.message
+        ? accountResponse.response.errors.message
+        : "";
+    const errorStatus =
+      accountResponse.response.errors && accountResponse.response.errors.code
+        ? accountResponse.response.errors.code
+        : "500";
+    throw new CustomError(
+      `failed to create/update group ${errMessage}`,
+      errorStatus
+    );
+  }
+  return accountResponse;
+};
+
+/*
+ * This functions is used for checking Number Data Types of payload.
+ * If the specified key is defined and it is not number then it throws error.
+ * @param {*} payload
+ * @returns
+ */
+const checkNumberDataType = payload => {
+  const numberAttributes = [
+    "territory_id",
+    "lead_source_id",
+    "owner_id",
+    "campaign_id",
+    "contact_status_id",
+    "lifecycle_stage_id",
+    "industry_type_id",
+    "business_type_id",
+    "number_of_employees",
+    "parent_sales_account_id"
+  ];
+  const errorAttributes = [];
+  numberAttributes.forEach(element => {
+    if (payload[element]) {
+      const value = payload[element];
+      if (!Number.isNaN(Number(value))) {
+        payload[element] = Number(value);
+      } else {
+        errorAttributes.push(element);
+      }
+    }
+  });
+  if (errorAttributes.length > 0) {
+    throw new CustomError(`${errorAttributes}: invalid number format`);
```

Comment:
```suggestion
    throw new CustomError(`${...errorAttributes}: invalid number format`);
```

Reply:
It is throwing an error `SyntaxError: Unexpected token ...`

Reply:
ok